oem/ibm/libpldmresponder/file_io.hpp

```diff
@@ -47,30 +72,198 @@ namespace fs = std::filesystem;
  */
 class DMA
 {
+    DMA(){}
   public:
+    
+    DMA(uint32_t length)
+    {
+        
+        responseReceived = false;
+        memAddr = nullptr;
+        xdmaFd = -1;
+        SourceFd = -1;
+        rc = 0;
+        iotPtr = nullptr;
+        iotPtrbc = nullptr;
+        timer = nullptr;
+        m_length = length;
+        static const size_t pageSize = getpagesize();
+        uint32_t numPages = m_length / pageSize;
+        pageAlignedLength = numPages * pageSize;
+        if (m_length > pageAlignedLength)
+        {
+            pageAlignedLength += pageSize;
+        }
+    }
+    ~DMA()
+    {
+        if (iotPtr != nullptr)
+        {
+            iotPtrbc = iotPtr.release();
+        }
+
+        if (iotPtrbc != nullptr)
+        {
+            delete iotPtrbc;
+            iotPtrbc = nullptr;
+        }
+
+        if (timer != nullptr)
+        {
+            auto time = timer.release();
+            delete time;
+        }
+
+        if (xdmaFd > 0)
+        {
+            close(xdmaFd);
+            xdmaFd = -1;
+        }
+        if (SourceFd > 0)
+        {
+            close(SourceFd);
+            SourceFd = -1;
+        }
+    }
+
+    int getDMAFd(bool nonBlokingMode, bool Reopensocket)
```

Comment:
add doxygen comment describing the method.

Reply:
done